src/Data/Managers/V2/GetCurrentUser/useGetCurrentUser.ts

```diff
@@ -0,0 +1,46 @@
+import { useQuery } from "@tanstack/react-query";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+
+interface LoggedUser {
+  data: {
+    username: string;
+  };
+}
+/**
+ * Custom hook for getting the current logged in user from the server.
+ *
+ * @returns {Query} - An object containing a custom hook to fetch user information.
+ */
+export const useGetCurrentUser = () => {
+  const { createHeaders, handleErrors } = useFetchHelpers();
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  const currentUserOrder = async (): Promise<LoggedUser> => {
+    const response = await fetch(baseUrl + `/api/v2/current_user/`, {
+      headers: createHeaders(),
+    });
+
+    await handleErrors(response);
+
+    return response.json();
+  };
+
+  return {
+    /**
+     * Custom hook to fetch the user information from the API once.
+     * @returns {UseQueryResult<LoggedUser, Error>} The result of the query, including the current user information.
+     */
+    useOneTime: () =>
```

Comment:
typing.
This is how I added the correct typing
 useOneTime: (): UseQueryResult<ServiceInstanceModel, Error> =>

Reply:
changed, I had to modify promise above though, because select property influence typing of the UseQueryResult